src/main/kotlin/ecommerce/repository/ProductJpaRepository.kt

```diff
@@ -0,0 +1,19 @@
+package ecommerce.repository
+
+import ecommerce.exception.ProductAlreadyInDBException
+import ecommerce.exception.ProductNotFoundException
+import ecommerce.model.Product
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.repository.findByIdOrNull
+
+fun ProductJpaRepository.getByIdOrThrow(id: Long): Product =
+    findByIdOrNull(id)
+        ?: throw ProductNotFoundException("Product not found")
+
+fun ProductJpaRepository.existsByNameOrThrow(name: String) {
+    if (existsByName(name)) throw ProductAlreadyInDBException("Product already exists with name: $name")
+}
+
+interface ProductJpaRepository : JpaRepository<Product, Long> {
+    fun existsByName(name: String): Boolean
+}
```

Comment:
I noticed that `MemberJpaRepository` returns `Member?` when looking for the member entity, while some other repositories like `ProductJpaRepository` handles not-found logic inside itself. I'm curious about this decision!

As for me, I would actually unify the way repositories handle not-found logic; for example, I would return `Product?` in this repository just like `MemberJpaRepository`, and keep it that way for all other ones.